### PR TITLE
Added initial support for mindy functions.

### PR DESCRIPTION
**Pull request by wvdschel**

### CODE REVIEW

File: melange-core/c-decl.dylan [hunk 1]
```diff
@@ -1134,7 +1134,7 @@ define method mapped-name
 end method mapped-name;
 
 define method type-name (decl :: <object-declaration>) => (result :: <string>);
-  decl.equated | decl.type.dylan-name;
+  decl.equated | decl.type.type-name;
```
Comment (waywardmonkeys) [inline melange-core/c-decl.dylan]: Can you tell me about why this change is needed? (And does C-FFI stuff still pass with it?)

**Comment by waywardmonkeys** [comment]
Apart from the one question (so far), could you update this with commit messages that are more in line with what I've been doing?

File: melange-core/c-decl.dylan (hunk 1 above)
Comment (wvdschel) [inline melange-core/c-decl.dylan]: 1. C-FFI still passes.
2. This change isn't needed, but I thought it was nicer to delegate to the member field's method for this instead of depending on its internal implementation directly.
